cognite/powerops/client/shop/cogshop_api.py

```diff
@@ -36,11 +50,324 @@ def auth(r: requests.PreparedRequest) -> requests.PreparedRequest:
+            end_time: End of time range SHOP is optimized over
+            shop_version: Version of SHOP to use (e.g. '16.0.2'or '15.7.0.0'). Required.
+
+            scenario_name: Name of the scenario. Required, does not have to be unique
+            model_name: Name of the model. Required, does not have to be unique
+            scenario_external_id: External ID of the scenario. Optional, must be unique
+            model_external_id: External ID of the model. Optional, must be unique
+            case_external_id: External ID of the SHOP case. Optional, must be unique
+        Returns:
+            ShopCaseWrite: A SHOP case that can be written to CDF
+        """
+        model_write = ShopModelWrite(
+            name=model_name,
+            shop_version=shop_version,
+            **({"external_id": model_external_id} if model_external_id else {}),
+        )
+
+        scenario_write = ShopScenarioWrite(
+            name=scenario_name,
+            model=model_write,
+            **({"external_id": scenario_external_id} if scenario_external_id else {}),
+        )
+
+        return self.prepare_shop_case_with_existing_scenario(
+            shop_file_list=shop_file_list,
+            start_time=start_time,
+            end_time=end_time,
+            shop_scenario_reference=scenario_write,
+            case_external_id=case_external_id,
+        )
+
+    def retrieve_shop_case(self, case_external_id: str) -> ShopCase:
+        """Retrieve a shop case from CDF"""
+        return self._po.shop_based_day_ahead_bid_process.shop_case.retrieve(external_id=case_external_id)
+
+    def list_shop_results_for_case(self, case_external_id: str, limit: int = 3) -> ShopResultList:
+        """
+        View the result of a SHOP case.
+        Args:
+            case_external_id: External ID of the SHOP case
+            limit: Number of results to return, -1 for all results
+        """
+        result_list: ShopResultList = self._po.shop_based_day_ahead_bid_process.shop_result.list(
+            case=case_external_id, limit=limit
+        )
+        return result_list
+
+    def retrieve_shop_result(self, result_external_id: str) -> ShopResult:
+        """Retrieve a shop result from CDF"""
+        return self._po.shop_based_day_ahead_bid_process.shop_result.retrieve(external_id=result_external_id)
+
     def list_shop_versions(self) -> list[str]:
         """List the available version of SHOP remotely  in CDF.
         Does not include versions in local version of CShaaS.
         SHOP releases should have the following format:
-        'SHOP-${{VERSION}}-pyshop-python{py_version}.linux.zip'
+        'SHOP-{VERSION}-pyshop-python{py_version}.linux.zip'
         """
         # todo? Add an endpoint to list the available versions of SHOP via powerops API?
         return [file.name for file in self._cdf.files.list(metadata={"shop:type": "shop-release"}, limit=-1)]
+
+    def retrieve_shop_case_graphql(self, case_external_id: str) -> ShopCase:

```

Comment:
Feels very strange to have this that does just the same as `retrieve_shop_case`.
If we keep it as a separate function, I would like the docstring to explain when it may be better to use this one (if the reason is something else than that the other one is unstable - if it is just for the instability, maybe use graphql as fallback inside `retrieve_shop_case` instead?)

Reply:
Updated docstring